src/App.js

```diff
@@ -25,44 +24,97 @@ const generateSquares = () => {
   return squares;
 }
 
-const App = () => {
+const updateCurrentPlayer = (currentPlayer, setCurrentPlayer) => {
+  const updatePlayer = currentPlayer === PLAYER_1 ? PLAYER_2:PLAYER_1
+  setCurrentPlayer(updatePlayer)
+}
 
-  // This starts state off as a 2D array of JS objects with
-  // empty value and unique ids.
+const App = () => {
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1)
+  const [winner, setWinner ] = useState(null)
 
-  // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  const checkForWinner = (squares) => {
+    let x = 0
+    let o = 0
+    let empty = false
+    const possibilities = [
+      [squares[0][0].value, squares[0][1].value, squares[0][2].value],
+      [squares[1][0].value, squares[1][1].value, squares[1][2].value],
+      [squares[2][0].value, squares[2][1].value, squares[2][2].value],
+      [squares[0][0].value, squares[1][0].value, squares[2][0].value],
+      [squares[0][1].value, squares[1][1].value, squares[2][1].value],
+      [squares[0][2].value, squares[1][2].value, squares[2][2].value],
+      [squares[0][0].value, squares[1][1].value, squares[2][2].value],
+      [squares[0][2].value, squares[1][1].value, squares[2][0].value]
+    ]
```

Comment:
This is an interesting variation to the data-driven approach of encoding the wining positions ahead of time. Notice that to make this as general as possible, it would still be helpful to take further step back and store the coordinates separately from the lookup step (the `[][].value` bit) but this could still be performed as an intermediate step.  That is: list of winning index combination → list of value triplets → process value triplets.